tools/verify_index_files/main/main.go

```diff
@@ -0,0 +1,251 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// This tool can be used to verify that the index files for multiple hosts that
+// share the same shards are the same. If they're not indentical, it will output
+// the name of any series that are missing, or have mismatched checksums. The tool
+// expects a directory where each subdirectory is the name of the host, and within
+// each of those subdirectories is the "data" directory for that host, exactly as
+// generated by M3DB itself. See the README for more details.
+
+package main
+
+import (
+	"flag"
+	"io"
+	"io/ioutil"
+	"log"
+	_ "net/http/pprof"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/m3db/m3db/persist/fs"
+	"github.com/m3db/m3db/tools"
+	"github.com/m3db/m3db/ts"
+	"github.com/m3db/m3x/pool"
+)
+
+var flagParser = flag.NewFlagSet("Verify Index files", flag.ExitOnError)
+
+type seriesChecksums struct {
+	host   string
+	shard  uint32
+	block  int64
+	series seriesMap
+}
+
+type seriesMap map[string]series
+
+type checksum uint32
+
+type series struct {
+	name     string
+	checksum checksum
+}
+
+var (
+	pathPrefixArg       = flagParser.String("path-prefix", "/var/lib/m3db", "Path prefix - must contain a folder called 'commitlogs'")
+	namespaceArg        = flagParser.String("namespace", "metrics", "Namespace")
+	shardsArg           = flagParser.String("shards", "", "Shards - set comma separated list of shards")
+	blocksArgs          = flagParser.String("blocks", "", "Start unix timestamp (Seconds) - set comma separated list of unix timestamps")
+	compareChecksumsArg = flagParser.Bool("compare-checksums", true, "Compare checksums")
+)
+
+var bytesPool pool.CheckedBytesPool
+
+func main() {
+	flagParser.Parse(os.Args[1:])
+
+	log.SetOutput(os.Stdout)
+	log.Println("Initializing bytes pool...")
+	bytesPool = tools.NewCheckedBytesPool()
+
+	var (
+		pathPrefix       = *pathPrefixArg
+		namespaceStr     = *namespaceArg
+		shardsVal        = *shardsArg
+		blocksVal        = *blocksArgs
+		compareChecksums = *compareChecksumsArg
+	)
+
+	blocks := parseBlockArgs(blocksVal)
+	hosts, err := ioutil.ReadDir(pathPrefix)
+	if err != nil {
+		log.Fatalf("Err reading dir: %s, err: %s\n", pathPrefix, err)
+	}
+
+	shards := parseShards(shardsVal)
```

Comment:
maybe someway to specify all shards would be useful?

Reply:
The bash script I wrote to SCP all the files only accepts a list of shards, so I'd have to fix that too :P